File: p049.py
import utils
import itertools

primes = set(filter(lambda n: n >= 1000, utils.primes(9999)))

def same_digits(x, y):
    return set(str(x)) == set(str(y))

for p in primes:
    for i in range(1, (10000 - p) / 2):
        pi = p + i
        pii = pi + i
        if pi in primes and pii in primes and same_digits(p, pi) and same_digits(p, pii):
            print str(p) + str(p + i) + str(p + i + i)

File: go/src/p049/p049.go
package main

import (
    "fmt"
    "utils"
)

func main() {
    primes := make(map[uint64]bool)
    for _, p := range utils.Primes(10000) {
        primes[p] = true
    }

    for p, _ := range primes {
        for i := uint64(1); p + i + i < 10000; i++ {
            if p == 1487 {
                continue
            }
            if primes[p + i] && primes[p + i + i] && sameDigits(p, p + i) && sameDigits(p, p + i + i) {
                fmt.Printf("%v%v%v\n", p, p + i, p + i + i)
                return
            }
        }
    }
}

func sameDigits(a, b uint64) (bool) {
    digits := make([]uint64, 10)

    for ; a > 0; a /= 10 {
        digits[a % 10]++
    }

    for ; b > 0; b /= 10 {
        digit := b % 10
        if digits[digit] <= 0 {
            return false
        } else {
            digits[digit]--
        }
    }

    return true
}
